File: leet_2319_x_matrix.cc
class Solution {
public:
    bool checkXMatrix(vector<vector<int>>& grid)
    {
        int len = grid.size();
        int r = -1, c;
        while (++r < len)
        {
            c = -1;
            while (++c < len)
            {
                if ((r == c || c == len - r - 1) && ! grid[r][c])
                    return false;
                else if (r ^ c && c ^ len - r - 1 && grid[r][c])
                    return false;
            }
        }
        return true;
    }
};

File: leet_2319_x_matrix.rs
impl Solution {
    pub fn check_x_matrix(grid: Vec<Vec<i32>>) -> bool {
        let mut len = grid.len();
        for r in 0 .. len {
            for c in 0 .. len {
                if c == r || c == len - r - 1 {
                    if grid[r][c] == 0 {
                        return false;
                    }
                } else {
                    if grid[r][c] != 0 {
                        return false;
                    }
                }
            }
        }
        return true;
    }
}
